core/src/main/scala/io/iteratee/Enumerator.scala

```diff
@@ -110,6 +110,29 @@ final object Enumerator {
       def empty: Enumerator[F, E] = Enumerator.empty
     }
 
+  private case class DeferEnumerator[F[_], E](build: () => Enumerator[F, E]) extends Enumerator[F, E] {
```

Comment:
unrelated, but useful. I thought we might wind up using it...

It may still be useful for stack safety, the trick of using a tailrec loop to evaluate a chain of these can be useful (it helps in making the ContT monad stack safe).